internal/api/controllers/private/runsCancelActions.go

```diff
@@ -0,0 +1,59 @@
+package private
+
+import (
+	"net/http"
+	"playbook-dispatcher/internal/api/controllers/public"
+	"playbook-dispatcher/internal/api/dispatch"
+	"playbook-dispatcher/internal/common/model/generic"
+	"playbook-dispatcher/internal/common/utils"
+
+	"github.com/google/uuid"
+)
+
+func CancelInputV2GenericMap(
+	cancelInput CancelInputV2,
+	account string,
+	runId *uuid.UUID,
+) generic.CancelInput {
+	orgIdString := string(cancelInput.OrgId)
+	principal := string(cancelInput.Principal)
+
+	result := generic.CancelInput{
+		Account:   account,
+		RunId:     runId,
+		OrgId:     &orgIdString,
+		Principal: &principal,
+	}
+
+	return result
+}
+
+func runCancelError(code int) *RunCanceled {
+	return &RunCanceled{
+		Code: code,
+		// TODO report error
+	}
+}
+
+func handleRunCancelError(err error) *RunCanceled {
+	if _, ok := err.(*dispatch.RunNotFoundError); ok {
+		return runCancelError(http.StatusNotFound)
+	}
+
+	if _, ok := err.(*dispatch.RecipientNotFoundError); ok {
+		return runCancelError(http.StatusConflict)
+	}
+
+	if _, ok := err.(*dispatch.RunCancelNotCancelableError); ok {
+		return runCancelError(http.StatusExpectationFailed)
+	}
+
+	return runCancelError(http.StatusInternalServerError)
+}
+
+func runCanceled(runID uuid.UUID) *RunCanceled {
+	return &RunCanceled{
+		Code:  http.StatusAccepted,
+		RunId: (*public.RunId)(utils.StringRef(runID.String())),
```

Comment:
this should not require a pointer (looks like you haven't re-generated the stubs after making the field required in the openapi schema)